azavea/tilegarden
# Leaflet Demo

## Overview

This PR adds a static webpage that uses Leaflet to demo the tiles drawn by Windshaft.


### Demo

![leaflet-demo](https://user-images.githubusercontent.com/16990679/41247125-0dff3ba8-6d7b-11e8-9cab-a022a1ebad63.png)

### Notes

From #12:
> Strictly speaking, this PR won't allow Windshaft to serve tiles properly as the example data set is not properly formatted, and altering the data seems outside the scope of this branch. In order to properly format the data, run './scripts/console db' to enter the database console and create a new column in the 'pa_gardens' table called 'the_geom_webmercator' that contains the values of the pre-existing 'geom' column re-projected to have a SRID of 3857. 
> ~~~~
>
> ALTER TABLE pa_gardens ADD COLUMN the_geom_webmercator geometry(POINT, 3857);
> UPDATE pa_gardens SET the_geom_webmercator = ST_Transform(ST_SetSRID(geom, 4326), 3857);
>
> ~~~~



## Testing Instructions

 * Run ./server/setup to build containers. All containers are built from images, but this is necessary in order to load in data to PostGIS.
 * Run ./scripts/server to start the containers (db, redis, windshaft, leaflet). Windshaft should tell you that map tiles are now being served.
 * Visit http://localhost/. You should see a map that displays the locations from the data set.


Resolves #2
